src/plots/cartesian/index.js

```diff
@@ -203,7 +203,7 @@ function plotOne(gd, plotinfo, cdSubplot, transitionOpts, makeOnCompleteCallback
     // remaining plot traces should also be able to do this. Once implemented,
     // we won't need this - which should sometimes be a big speedup.
     if(plotinfo.plot) {
-        plotinfo.plot.selectAll('g:not(.scatterlayer)').selectAll('g.trace').remove();
+        plotinfo.plot.selectAll('g:not(.scatterlayer):not(.ohlclayer)').selectAll('g.trace').remove();
```

Comment:
This query is getting ridiculous. 

Can we instead loop over [`constants.traceLayerClasses`](https://github.com/plotly/plotly.js/blob/18e19a86a3e683c1d11f6f4a04eb7f89d1b6108f/src/plots/cartesian/constants.js#L64)?

Reply:
> Can we instead loop over `constants.traceLayerClasses`?

Somehow we need to distinguish between trace types that know how to update and those that don't - I feel like rather than adding another category for this, we should just teach the remaining types how to update. Probably actually not that hard to do, right? Just adding the right `exit()` statements and moving things appropriately between `enter()` and `each`?

I wouldn't want to do that in this PR of course...

Reply:
Moved to https://github.com/plotly/plotly.js/issues/2563 

Thanks!